File: src/Reporting.Core/Models/ReportAdminDetailsModel.cs
﻿namespace Reporting.Core.Models
{
    using System;
    using System.Collections.Generic;

    public class ReportAdminDetailsModel
    {
        public string? Key { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public bool IsActive { get; set; }
        public bool HasParameters { get; set; }
        public int ReportSourceId { get; set; }
        public ReportSourceModel? ReportSource { get; set; }
        public IEnumerable<ReportParameterModel>? Parameters { get; set; }
        public IEnumerable<ReportColumnDefinitionModel> ColumnDefinitions { get; set; } = Array.Empty<ReportColumnDefinitionModel>();
        public string? CreatedByUser { get; set; }
        public DateTime CreatedAtDate { get; set; }
        public string? LastUpdatedByUser { get; set; } = null;
        public DateTime? LastUpdatedAtDate { get; set; } = null;
    }
}

File: src/reporting-app/src/models/report-admin-details-model.ts
import { ReportSourceModel } from './report-source-model';
import { ReportParameterModel } from './report-parameter-model';
import { ReportColumnDefinitionModel } from './report-column-definition-model';

export interface ReportAdminDetailsModel {
  key?: string;
  name?: string;
  description?: string;
  isActive: boolean;
  hasParameters: boolean;
  reportSourceId: number;
  reportSource: ReportSourceModel;
  parameters?: ReportParameterModel[];
  columnDefinitions?: ReportColumnDefinitionModel[];
  createdByUser?: string;
  createdAtDate: string;
  lastUpdatedByUser?: string;
  lastUpdatedAtDate?: string;
}
